agent/ebs/watcher.go

```diff
@@ -92,6 +93,94 @@ func (w *EBSWatcher) Start() {
 	}
 }
 
+// Method to handle watcher's tick.
+// If there are no pending EBS volume attachments in agent state, then this method is a no-op.
+// If there are pending EBS volume attachments in agent state then this method will ensure
+// that EBS Managed Daemon is running and then scan the host for the EBS volumes and process
+// the ones that are found.
+func (w *EBSWatcher) tick() {
+	pendingEBS := w.agentState.GetAllPendingEBSAttachmentsWithKey()
+	if len(pendingEBS) <= 0 {
+		return
+	}
+	if !w.daemonRunning() {
+		log.Info("EBS Managed Daemon is not currently running. Skipping EBS Watcher tick.")
+		return
+	}
+	foundVolumes := apiebs.ScanEBSVolumes(pendingEBS, w.discoveryClient)
+	w.overrideDeviceName(foundVolumes)
+	w.StageAll(foundVolumes)
+	w.NotifyAttached(foundVolumes)
+}
+
+// Checks if EBS Daemon Task is running and starts a new one it if it's not.
+func (w *EBSWatcher) daemonRunning() bool {
+	csiTask := w.taskEngine.GetDaemonTask(md.EbsCsiDriver)
+
+	// Check if task is running or about to run
+	if csiTask != nil && csiTask.GetKnownStatus() == status.TaskRunning {
+		logger.Debug("EBS Managed Daemon is running", logger.Fields{field.TaskID: csiTask.GetID()})
+		return true
+	}
+	if csiTask != nil && csiTask.GetKnownStatus() < status.TaskRunning {
+		logger.Debug("EBS Managed Daemon task is pending transitioning to running", logger.Fields{
+			field.TaskID:      csiTask.GetID(),
+			field.KnownStatus: csiTask.GetKnownStatus(),
+		})
+		return false
+	}
+
+	// Task is not running or about to run, we need to start a new one.
+
+	// Log some information.
```

Comment:
nit is this comment necessary (probably a leftover?)

Reply:
Sure. I will update.

Reply:
Updated in https://github.com/aws/amazon-ecs-agent/pull/3984/commits/a56ea6195b4378a019d0016a40996720c4ece7a1